packages/terra-clinical-modal-manager/src/components/ModalManager.jsx

```diff
@@ -0,0 +1,198 @@
+    // renderings that occur.
+    this.forceFullscreenModal = false;
+
+    this.updateFullscreenState = this.updateFullscreenState.bind(this);
+    this.buildModalComponents = this.buildModalComponents.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateFullscreenState();
+    window.addEventListener('resize', this.updateFullscreenState);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateFullscreenState);
+  }
+
+  updateFullscreenState() {
+    const previousFullscreenState = this.forceFullscreenModal;
+
+    this.forceFullscreenModal = window.innerWidth < getBreakpoints().small;
+
+    // Only update the modal if it's minimized, open, and changing states.
+    if (!this.props.isMaximized && this.props.isOpen && previousFullscreenState !== this.forceFullscreenModal) {
+      this.forceUpdate();
+    }
+  }
+
+  buildModalComponents() {
+    if (!this.props.modalComponentKeys || !this.props.modalComponentKeys.length) {
+      return null;
+    }
+
+    return this.props.modalComponentKeys.map((componentKey, index) => {
+      const componentData = this.props.modalComponentData[componentKey];
+
+      const ComponentClass = AppDelegate.getComponentForDisclosure(componentData.name);
+
+      if (!ComponentClass) {
+        return undefined;
+      }
+
+      const appDelegate = AppDelegate.create({
+        disclose: (data) => {
+          this.props.pushModal(data);
+        },
+        dismiss: (index > 0 ?
+          (data) => {
+            this.props.popModal(data);
+          } :
+          (data) => {
+            this.props.closeModal(data);
+          }
+        ),
+        closeDisclosure: () => { this.props.closeModal(); },
+        goBack: index > 0 ? () => { this.props.popModal(); } : null,
+        maximize: !this.props.isMaximized ? () => { this.props.maximizeModal(); } : null,
+        minimize: this.props.isMaximized ? () => { this.props.minimizeModal(); } : null,
+      });
+
+      return <ComponentClass key={componentKey} {...componentData.props} app={appDelegate} />;
+    });
+  }
+
+  render() {
+    const { app, openModal, closeModal, size, isOpen, isMaximized, children } = this.props;
+
+    const sizeClass = `terraClinical-ModalManager-modal--${size}`;
+
+    const modalClassNames = classNames([
+      'terraClinical-ModalManager-modal',
+      { [sizeClass]: !(isMaximized || this.forceFullscreenModal) },
+    ]);
+
+    return (
+      <div className="terraClinical-ModalManager">
+        {React.Children.map(children, (child) => {
+          const childAppDelegate = AppDelegate.clone(app, {
+            disclose: (data) => {
+              if (data.preferredType === 'modal' || !app) {
+                openModal(data);
+              } else {
+                app.disclose(data);
+              }
+            },
+          });
+
+          return React.cloneElement(child, { app: childAppDelegate });
+        })}
```

Comment:
Should this function be broken out for readability? Then this could be reduced to {clonedChildren}